Collector/ProfileFactory.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Http\HttplugBundle\Collector;
+
+/**
+ * @author Fabien Bourigault <bourigaultfabien@gmail.com>
+ *
+ * @internal
+ */
+class ProfileFactory
```

Comment:
If I mock the factories, I'm no loner able to have control on the `Stack` or `Profile` objects during tests :/
Maybe my tests over describe the plugin behaviors.